srcs/runners_manager/vm_creation/gcloud/GcloudManager.py

```diff
@@ -73,6 +74,16 @@ def configure_instance(
         )
         disk_size_gb = runner.vm_type.config["disk_size_gb"]
         disk_type = f"projects/{self.project_id}/zones/{self.zone}/diskTypes/pd-ssd"
+
+        automatic_restart = True
+        provisioning_model = "STANDARD"
+        preemptible = False
+        if "preemptible" in runner.vm_type.config:
+            preemptible = bool(runner.vm_type.config["preemptible"])
+            if preemptible:
+                provisioning_model = "SPOT"
+                automatic_restart = False
```

Comment:
Can we have all those settings in `runner.vm_type.config` with the appropriate default values?

Reply:
Yes we can but I'm not sure it's a good idea. `automatic_restart` is False when it's a `SPOT` instance, `preemptible` is True when it's a SPOT instance. In the other hand, when it's a `DEFAULT` instance, `preemptible` is False and `automatic_restart` is True. So, it's a cause-and-effect relationship. Might be bette to just fill in the information once with `preemptible`, right?

Reply:
For me, a SPOT instance is a new version of preemptible instances. I couldn't find the exact answer regarding what would happen if we removed the `preemptible` tag, but I thought I would directly test it with the Google Cloud Console by creating a Spot VM and only specifying "Spot". So, I conducted the experiment, and once the VM was created, I went to verify, and the preemptible variable was set to "on". I believe the two are linked.

I will test in my code to enable one and not the other to see what it will result in.

Reply:
![Uploading Capture d’écran 2023-06-09 à 10.21.43.png…]()

Reply:
<img width="450" alt="Capture d’écran 2023-06-09 à 10 21 43" src="https://github.com/scality/runner-manager/assets/44836297/46931b1a-f354-4e04-b8e4-a9762d872925">

Reply:
`[ERROR] runner_manager: 400 POST https://compute.googleapis.com/compute/v1/projects/scality-prod-ga-runners/zones/us-east1-c/instances: Invalid value for field 'resource.scheduling.provisioningModel': 'SPOT'. preemptible=false and provisioning_model=SPOT is contradicting.`

- If I set the `preemptible` tag to `False` while it is a `SPOT` instance, it returns an error that tells us we are contradicting ourselves. There is indeed a cause-and-effect relationship.

Reply:
ok cool, thanks for checking, so it confirms we need the `preemtible` flag as well.